docs/gallery/howto/autogen/control-flow.py

```diff
@@ -0,0 +1,283 @@
+# In addition, the ``if`` conditional can also be expressed using the ``@task.graph`` decorator.
+# So let's dive right into it!
+
+# %%
+# Setting up the AiiDA environment
+# --------------------------------
+# First, we need to set up our AiiDA environment, importing the necessary entities from ``aiida-workgraph`` and loading the AiiDA profile.
+
+from aiida_workgraph import WorkGraph, task, If, While
+from aiida_workgraph.utils import generate_node_graph
+from aiida import load_profile
+
+load_profile()
+
+
+# %%
+# If
+# ==
+
+# %%
+# Workflow description
+# --------------------
+#
+# Now, let's see how we can convert the following arithmetic workflow containing an ``if`` conditional into a WorkGraph:
+#
+# .. code-block:: python
+#
+#    # First addition
+#    result = 1 + 1
+#
+#    # Conditionally execute addition or multiplication
+#    if result < 0:
+#        result = result + 2
+#    else:
+#        result = result * 3
+#
+#    # Last addition
+#    result = result + 1
+
+# %%
+# First, we define the relevant arithmetic operations as WorkGraph tasks.
+
+
+@task
+def add(x, y):
+    return x + y
+
+
+@task
+def multiply(x, y):
+    return x * y
+
+
+# Those will present the processes executed in the workflow, such that their provenance is tracked.
+
+# %%
+# Context manager
+# ---------------
+#
+# To define the conditional logic of the workflow, WorkGraph provides an ``If`` context manager.
+# Using the ``with If`` block, all the child tasks are automatically encapsulated.
+# Their execution when the workflow is run is based on the defined conditions.
+
+with WorkGraph("if_context_manager") as wg:
+
+    add1_outputs = add(x=1, y=1)
+
+    with If(add1_outputs.result < 0):
+        cond_add_outputs = add(x=add1_outputs.result, y=2)
+        wg.ctx.result = cond_add_outputs.result
+
+    with If(add1_outputs.result >= 0):
+        cond_mult_outputs = multiply(x=add1_outputs.result, y=3)
+        wg.ctx.result = cond_mult_outputs.result
+
+    final_outputs = add(x=wg.ctx.result, y=1)
+
+    # In the following, we have to use the ``<<`` syntax to explicitly tell WG to wait
+    # with the execution of the last task until the previous two tasks have finished.
+    # In principle, as we are dealing with python functions that are run in a blocking
+    # manner here, the example would also work without ``<<``. However, if the tasks
+    # would be submitted to the deamon in a non-blocking fashion, the explicit waiting
+    # enforced by ``<<`` is strictly required, so we also apply it here for consistency
+    # and correctness.
+    final_outputs << cond_add_outputs
+    final_outputs << cond_mult_outputs
+
+    # Finally, we set the result of the last task as the global workflow output result
+    wg.outputs.result = final_outputs.result
+
+wg.run()
+print(f"State of WorkGraph: {wg.state}")
+print(f"Result            : {wg.outputs.result.value}")
+assert wg.outputs.result.value == 7
+
```

Comment:
You may also add a dedicated section to explain this, because it is used in the `While` case too. E.g, add a note:

WorkGraph doesn't track context variables (wg.ctx) for automatic dependency resolution because they could introduce cyclical dependencies between tasks.
By using the `<<` or `>>` operators, we explicitly declare that the `final_outputs` task must wait for both `cond_add_outputs` and `cond_mult_outputs` to finish before running, ensuring correct ordering when reading `wg.ctx.result`.